lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,193 @@
+      pool_id = pool_cmdret[:stdout].split("\n").first.to_s.strip
+      perform_agent_commands(deployment_master, user, ["sudo subscription-manager attach --pool='#{pool_id}'"]) unless pool_id.blank?
+      enabled_repos_cmdret = perform_agent_commands(deployment_master, user, ['sudo subscription-manager repos --disable="*"',
+                                                                              "sudo subscription-manager repos --enable=#{SUBSCRIPTION_REPOS.join(" --enable=")}",
+                                                                              "sudo subscription-manager repos --list-enabled"])
+      enabled_repos = enabled_repos_cmdret[:stdout].split(" ").to_a.map(&:strip).reject(&:empty?)
+      missing_repos = SUBSCRIPTION_REPOS - enabled_repos
+      raise StandardError, "couldn't register to following repos : #{missing_repos.join(',')} with SKU : #{rhsm_sku}" unless missing_repos.empty?
+    end
+
+    def subscribe_cluster(deployment_master, user)
+      perform_agent_commands(deployment_master, user, ["sudo mv ~/rhel_subscribe_inventory.yaml #{INVENTORIES_PATH}"])
+      run_playbook_command(deployment_master, user, "ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/rhel_subscribe.yml -i "\
+                         "#{INVENTORIES_PATH}rhel_subscribe_inventory.yaml 1> #{ANSIBLE_LOG} 2> #{ANSIBLE_ERROR_LOG} < /dev/null")
+    end
+
+    def nodes_subscription_needed?
+      container_deployment_nodes.count > 1
+    end
+
+    def ssh_user
+      ssh_auth.userid
+    end
+
+    def add_deployment_provider(options)
+      object_send(:add_deployment_provider, options)
+    end
+
+    def regenerate_ansible_inventory
+      object_send(:generate_ansible_yaml)
+    end
+
+    def regenerate_ansible_subscription_inventory
+      object_send(:generate_ansible_inventory_for_subscription)
+    end
+
+    def add_automation_task(task)
+      ar_method do
+        @object.automation_task = AutomationTask.find_by_id(task.id)
+        @object.save!
+      end
+    end
+
+    def customize(options = {})
+      ar_method do
+        if @object.customizations.empty?
+          @object.customizations = {:agent => {}}
+        end
+        @object.customizations[:agent].merge!(options)
+        @object.save!
+      end
+    end
+
+    def perform_scp(ip, username, local_path, remote_path)
+      Net::SCP.upload!(ip, username, local_path, remote_path, :ssh => {:key_data => ssh_auth.auth_key})
+    end
+
+    def perform_agent_commands(ip, username, commands = [])
+      @ssh ||= LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      @ssh.perform_commands(commands)
+    end
+
+    def check_connection(deployment_host_ip, username)
+      agent = LinuxAdmin::SSHAgent.new(ssh_auth.auth_key, "#{SSH_AGENT_PATH}ssh_manageiq_#{id}")
+      nodes_ips = container_deployment_nodes.collect(&:node_address)
+      nodes_ips.delete(deployment_host_ip)
+      @ssh ||= LinuxAdmin::SSH.new(deployment_host_ip, username)
```

Comment:
this doesn't work for me, you forgot ssh key, also it fails below if it's all in one

I did this:

```
@ssh ||= LinuxAdmin::SSH.new(deployment_host_ip, username, ssh_auth.auth_key)
raise StandardError, "couldn't connect to : #{deployment_host_ip}" if @ssh.perform_commands(["echo 0"])[:stdout].exclude?("0\r\n")
return true if nodes_ips.blank?
```